src/controller/java/CHIPDeviceController-JNI.cpp

```diff
@@ -545,6 +545,55 @@ JNI_METHOD(void, setAttestationTrustStoreDelegate)
     }
 }
 
+JNI_METHOD(void, startOTAProvider)(JNIEnv * env, jobject self, jlong handle, jobject otaProviderDelegate)
+{
+#if CHIP_DEVICE_CONFIG_DYNAMIC_SERVER
+    chip::DeviceLayer::StackLock lock;
+    CHIP_ERROR err                           = CHIP_NO_ERROR;
+    AndroidDeviceControllerWrapper * wrapper = AndroidDeviceControllerWrapper::FromJNIHandle(handle);
+
+    VerifyOrReturn(wrapper != nullptr, ChipLogError(Controller, "wrapper is null"));
+
+    ChipLogProgress(Controller, "startOTAProvider() called");
+
+    if (otaProviderDelegate != nullptr)
+    {
+        jobject otaProviderDelegateRef = env->NewGlobalRef(otaProviderDelegate);
```

Comment:
where do we release this otaProviderDelegateRef?

Reply:
why do you think we need global ref for this delegate? In invoke and write, we also pass callback via jobject, and they are async operations, it seems the jobject passed from java don't need to be global?

Reply:
I suggest to have JniGlobalReference mOtaProviderDelegate in OTAProviderDelegate, which is recommended pattern to handle global ref for class member.
see example from https://github.com/project-chip/connectedhomeip/pull/29710/files

Reply:
I modify.